utils/utils.py

```diff
@@ -71,3 +73,22 @@ def dump_log(config, metrics, split):
         result[split] = [metrics]
     with open(log_path, 'w') as fp:
         json.dump(result, fp)
+
+
+def dump_top_k_prediction(config, classes, y_pred, k=100):
+    """Dump top k prediction to the predict_path. The format of this file is:
+    <label1>:<value1> <label2>:<value2> ...
+    """
+
+    if config.predict_path:
+        predict_path = config.predict_path
+    else:
+        predict_path = os.path.join(config.result_dir, config.run_name, 'prediction.txt')
+    
+    os.makedirs(os.path.dirname(predict_path), exist_ok=True)
+    print(f'Dump top {k} prediction to {predict_path}.')
+    with open(predict_path, 'w') as fp:
+        for pred in np.vstack(y_pred):
```

Comment:
Why we need vstack here?

Reply:
Oh! That a long story ... We do `add_batch` in `MultiLabelMetrics`. I will check this again.